### PR TITLE
[UI] Text classifier: annotation task interaction enhancement

### LINKED ISSUE #552: UX-UI Annotation task : interaction enhancement ¨Next release¨
### New buttons styles

Annotation view (with/without score)
- Predicted label
- Annotated label
- Default label
- Hover state
- Active state

Explore view (with/without score)
- Predicted label
- Annotated label
- Default label

### Define responsive grid
Max width or column number (according to the design chosen)

### Stop moving label tag 
- Fix label position
- Implement logic order to show selected label when it's hidden in "+"

### PR DESCRIPTION
**Pull request by leiyre**
fix #552

### CODE REVIEW

File: frontend/components/text-classifier/results/ClassifierAnnotationArea.vue [hunk 1]
```diff
@@ -97,14 +61,18 @@ export default {
...
+    maxLabels: 7,
+    testA: false,
   }),
   computed: {
+    maxLabelsShown() {
+      if (this.selectedLabels.length > this.maxLabels) {
+        return this.selectedLabels.length
+      } else {
+        return this.maxLabels
+      }
```
Comment (frascuchon) [inline frontend/components/text-classifier/results/ClassifierAnnotationArea.vue]:
```suggestion
      return Math.max(this.selectedLabels.length, this.maxLabels); 
```

File: frontend/components/text-classifier/results/ClassifierAnnotationArea.vue [hunk 2]
```diff
@@ -181,103 +161,59 @@ export default {
...
 }
 .feedback-interactions {
   margin: 1.5em auto 0 auto;
   padding-right: 0;
-  & > div {
-    width: 100%;
-  }
+  // & > div {
+  //   width: 100%;
+  // }
```
Comment (frascuchon) [inline frontend/components/text-classifier/results/ClassifierAnnotationArea.vue:213]:
```suggestion
```

File: frontend/components/text-classifier/results/ClassifierAnnotationArea.vue [hunk 3]
```diff
@@ -181,103 +161,59 @@ export default {
...
   &__items {
-    display: flex;
-    flex-flow: wrap;
-    margin-left: -1%;
-    margin-right: -1%;
+    // display: flex;
+    // flex-flow: wrap;
+    // margin-left: -1%;
+    // margin-right: -1%;
```
Comment (frascuchon) [inline frontend/components/text-classifier/results/ClassifierAnnotationArea.vue:218]:
```suggestion
```

File: frontend/components/text-classifier/results/ClassifierAnnotationButton.vue [hunk 4]
```diff
@@ -184,7 +184,8 @@ $annotation-button-touch-size: 48px;
       overflow: hidden;
       transition: transform 0.3s ease;
       &__text {
-        max-width: calc(100% - 10px);
+        // max-width: calc(100% - 10px);
```
Comment (frascuchon) [inline frontend/components/text-classifier/results/ClassifierAnnotationButton.vue]:
```suggestion
```

File: frontend/components/text-classifier/results/ClassifierAnnotationArea.vue [hunk 5]
```diff
@@ -97,14 +61,18 @@ export default {
   },
   data: () => ({
     searchText: undefined,
-    componentLabels: undefined,
-    maxLabelsShown: 12,
-    selectedLabel: undefined,
-    dropdownLabels: undefined,
-    visible: undefined,
     selectedLabels: [],
+    maxLabels: 7,
```
Comment (frascuchon) [inline frontend/components/text-classifier/results/ClassifierAnnotationArea.vue]: Could we centralize the max label constant defintion ? 

something like:

```javascript
import { TextClassification } from "somewhere";  
Textclassification.MAX_VISIBLE_LABELS;
```

File: frontend/components/text-classifier/results/ClassifierExplorationArea.vue [hunk 6]
```diff
@@ -30,24 +39,63 @@ export default {
     }
   },
   data: () => ({
-    maxLabels: 16,
+    searchText: undefined,
+    maxLabels: 7,
```
Comment (frascuchon) [inline frontend/components/text-classifier/results/ClassifierExplorationArea.vue]: Here, we can user `MAX_VISIBLE_LABELS` constant when defined.

File: frontend/components/text-classifier/results/ClassifierExplorationArea.vue [hunk 7]
```diff
@@ -30,24 +39,63 @@ export default {
...
+        label.class.toLowerCase().match(this.searchText)
+      );
+    },
   },
+  methods: {
+    showHiddenLabels() {
+      this.maxLabels = this.filteredLabels.length;
+    },
+    hideHiddenLabels() {
+      this.maxLabels = 7;
```
Comment (frascuchon) [inline frontend/components/text-classifier/results/ClassifierExplorationArea.vue]: Ditto

**Review by frascuchon — CHANGES_REQUESTED** [review]

File: frontend/components/text-classifier/results/ClassifierAnnotationArea.vue (hunk 2 above)
Comment (leiyre) [reply frontend/components/text-classifier/results/ClassifierAnnotationArea.vue:213]: @frascuchon waiting to choose a style for the labes (fix width or natural width) to eliminate it

File: frontend/components/text-classifier/results/ClassifierAnnotationArea.vue (hunk 3 above)
Comment (leiyre) [reply frontend/components/text-classifier/results/ClassifierAnnotationArea.vue:218]: @frascuchon waiting to choose a style for the labes (fix width or natural width) to eliminate it

**Review by frascuchon — APPROVED** [review]

**Pull request merged**